src/core_tests/replay_flag.rs

```diff
@@ -0,0 +1,68 @@
+use crate::{
+    machines::{test_help::TestHistoryBuilder, WorkflowMachines},
+    protos::coresdk::workflow_commands::StartTimer,
+    protos::temporal::api::enums::v1::CommandType,
+    test_help::canned_histories,
+    test_workflow_driver::{CommandSender, TestWorkflowDriver},
+    workflow::WorkflowManager,
+};
+use rstest::{fixture, rstest};
+use std::time::Duration;
+
+// TODO: Dedupe
+#[fixture]
+fn fire_happy_hist() -> (TestHistoryBuilder, WorkflowMachines) {
+    let twd = TestWorkflowDriver::new(|mut command_sink: CommandSender| async move {
+        let timer = StartTimer {
+            timer_id: "timer1".to_string(),
+            start_to_fire_timeout: Some(Duration::from_secs(5).into()),
+        };
+        command_sink.timer(timer).await;
+        command_sink.complete_workflow_execution();
+    });
+
+    let t = canned_histories::single_timer("timer1");
+    let state_machines = WorkflowMachines::new(
+        "wfid".to_string(),
+        "runid".to_string(),
+        Box::new(twd).into(),
+    );
+
+    assert_eq!(2, t.as_history().get_workflow_task_count(None).unwrap());
+    (t, state_machines)
+}
+
+#[rstest]
+fn replay_flag_is_correct(fire_happy_hist: (TestHistoryBuilder, WorkflowMachines)) {
```

Comment:
I get what this test does but not why it works.
Maybe add a comment about the history and the state of the state machine in this test?

Reply:
added